tests/test_cache.py

```diff
@@ -195,6 +200,138 @@ def test_cache_using_string(self):
+        TestCache.test_list = []
+        self.assertListEqual(self.get_cache_empty(TestCache.test_list), [])
+        TestCache.test_list = ['value', '100']
+        self.assertListEqual(self.get_cache_empty(TestCache.test_list), [])
+        with freeze_time(datetime.utcnow() + timedelta(seconds=2)):
+            self.assertListEqual(self.get_cache_empty(TestCache.test_list), ['value', '100'])
+        TestCache.test_list = []
+        self.assertListEqual(self.get_cache_empty(TestCache.test_list), ['value', '100'])
+        with freeze_time(datetime.utcnow() + timedelta(seconds=4)):
+            self.assertListEqual(self.get_cache_empty(TestCache.test_list), [])
+
+        # String
+        self.clear_cache_empty()
+        TestCache.test_string = ""
+        self.assertEqual(self.get_cache_empty(TestCache.test_string), "")
+        TestCache.test_string = "Hello World"
+        self.assertEqual(self.get_cache_empty(TestCache.test_string), "")
+        with freeze_time(datetime.utcnow() + timedelta(seconds=2)):
+            self.assertEqual(self.get_cache_empty(TestCache.test_string), "Hello World")
+        TestCache.test_string = ""
+        self.assertEqual(self.get_cache_empty(TestCache.test_string), "Hello World")
+        with freeze_time(datetime.utcnow() + timedelta(seconds=4)):
+            self.assertEqual(self.get_cache_empty(TestCache.test_string), "")
+
+        # Set
+        self.clear_cache_empty()
+        TestCache.test_set = set()
+        self.assertSetEqual(self.get_cache_empty(TestCache.test_set), set())
+        TestCache.test_set = {1.0, "Hello", (1, 2, 3)}
+        self.assertSetEqual(self.get_cache_empty(TestCache.test_set), set())
+        with freeze_time(datetime.utcnow() + timedelta(seconds=2)):
+            self.assertSetEqual(self.get_cache_empty(TestCache.test_set), {1.0, "Hello", (1, 2, 3)})
+        TestCache.test_set = set()
+        self.assertSetEqual(self.get_cache_empty(TestCache.test_set), {1.0, "Hello", (1, 2, 3)})
+        with freeze_time(datetime.utcnow() + timedelta(seconds=4)):
+            self.assertSetEqual(self.get_cache_empty(TestCache.test_set), set())
+
+    def test_cache_empty_false(self):
+        # Dict
+        self.clear_cache_empty_false()
+        TestCache.test_dict = {}
+        self.assertDictEqual(self.get_cache_empty_false(TestCache.test_dict), {})
+        TestCache.test_dict = {'value': '100'}
+        self.assertDictEqual(self.get_cache_empty_false(TestCache.test_dict), {'value': '100'})
+        TestCache.test_dict = {}
+        self.assertDictEqual(self.get_cache_empty_false(TestCache.test_dict), {'value': '100'})
+        with freeze_time(datetime.utcnow() + timedelta(seconds=2)):
+            self.assertDictEqual(self.get_cache_empty_false(TestCache.test_dict), {})
+        TestCache.test_dict = {'value': '100'}
+        self.assertDictEqual(self.get_cache_empty_false(TestCache.test_dict), {'value': '100'})
+
+        # Tuple
+        self.clear_cache_empty_false()

```

Comment:
this is hard to review. 
`test_cache_empty_false` + `test_cache_empty` can be done the same as the test of result to dict. 
just create one function for testing with cache and another function to test without cache. 
and pass diffrent value every time

then just pass a different variable every time